plugins/mcreator-core/blockly/js/extensions_procedure_deps.js

```diff
@@ -25,20 +25,15 @@ Blockly.Extensions.register('procedure_dependencies_onchange_mixin',
             const group = Blockly.Events.getGroup();
             // Makes it so the block change and the unplug event get undone together.
             Blockly.Events.setGroup(changeEvent.group);
-            let valid = true;
             const procedure = this.getFieldValue('procedure');
             for (var i = 0; this.getField('name' + i); i++) {
                 const prevType = this.getInput('arg' + i).connection.getCheck();
+                let depType = javabridge.getDependencyType(procedure, this.getFieldValue('name' + i));
+                if (!depType && procedure !== '')
+                    depType = [];

```

Comment:
Alternatively, we can set it to `'Null'` instead given this type is already in use by Blockly blocks. (Depends on how appropriate it would look, though.)

Reply:
I'd say it's fine to use the empty array.

I'd personally make `getDependencyType` return the empty string if dependency exists, but no variable type is associated to it. With the current implementation, blocks are unplugged immediately while renaming a dependency.

Reply:
> With the current implementation, blocks are unplugged immediately while renaming a dependency.

Should that behavior be changed?

Reply:
Asking because I'm not sure if we can check whether the field is being edited at a given moment.

Reply:
Current behavior is fine

Reply:
Then this is resolved?

Reply:
Yes